### PR TITLE
DPC-1257: New account creation page

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-1257](https://jira.cms.gov/browse/DPC-1257)

Using layout from H-team, create the user account registration set up and registration page.

### Proposed Changes

- Users should input: first name, last name, email, password, and name of implementer organization.
- Emails should be checked with existing rules at time of submission.
- Users should check a box to agree to the ToS (with a link to the ToS).
- Leave a space for language to gently redirect non-implementers.
- Confirm page meets accessibility requirements.

### Change Details

- Add GUI for password requirements
- Add styling for registration form
- Install devise
- Set up devise notices/errors
- Install truemail
- Tests for user registration and user model

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

- LGTM is throwing an error for the according JS file that seems to be from a CMS approved source.
- Code climate is mad about a lot of things. Some of which I _can_ change (like the single vs. double quotes) and some I can't (how many lines some of the files are, like the schema), but none of it prevents the rails app from working nor does it improve or impede anything.

Screen Shot 2021-04-13 at 3 35 59 PM

Screen Shot 2021-04-13 at 3 36 43 PM

Screen Shot 2021-04-13 at 3 59 49 PM

Screen Shot 2021-04-13 at 4 57 02 PM

### CODE REVIEW

File: dpc-impl/config/credentials.yml.enc [hunk 1]
```diff
@@ -0,0 +1 @@
+kTQlXp1rQE6REcG4r/iBHGlPEUKOdP9qr2aNVGHTi9xq/ZkHfSh0xhVn4iImCN0/NxTPewQ8/FPZ0IOx0GMaH2fngGHiiuYsi+hdZbjO+s1owIf8pDzhYMvaN9bHDsDeG4aUbCRh0CBbfCW+1TOPz6aoJGitMjrTZ8U5ti86D7oWGrKeYVF0AdTw+4irSqSi1G13LS/VErwjNLdJ/+Vz7EgHCaWhHXZbs5s6K5vdhrfZ8ulJuVo6Z11LmJ0sNnX2B3SnVlVQ53xCIs3qVPoJmWpnpLcI8ZwRv1xcFTXtFxy4Ak2HtWTJ9BQD+m9spo0EwVXNE1jLywfJ8dZv5y2DA6K1vOWfhlWhoFyXJD74jQVphRhyRhqY3SdAax3Ut8UiPyxpN/7ZkoLa7a9eKSgzKFC4Saw075vNKlEk--Ov9w7jyrEUHrdmqh--XEVLt+I52UcGmLxPRyoocA==
```
Comment (dhgreene) [inline dpc-impl/config/credentials.yml.enc:1]: Shall we delete this?  😄

**Review by dhgreene — COMMENTED** [review]
I assume the commented lines in the controllers are still a work in progress.  It's coming along well!

File: dpc-impl/config/credentials.yml.enc (hunk 1 above)
Comment (Sun-Mountain) [reply dpc-impl/config/credentials.yml.enc:1]: Oh, yeah!

File: dpc-impl/spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    end
+  end
+
+  describe '#name' do
+    it 'returns the full name' do
+      expect(subject.name).to eq("#{subject.first_name} #{subject.last_name}")
+    end
+  end
+
+  describe '#implementer' do
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/spec/models/user_spec.rb:32]: I'm also fine either way.

[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply dpc-impl/spec/models/user_spec.rb:32]: I'm also fine either way, but google autocorrects `implementor` to `implementer`

Screen Shot 2021-04-14 at 10 36 03 AM

**Comment by Sun-Mountain** [comment]
> I assume the commented lines in the controllers are still a work in progress. It's coming along well!

@dhgreene those are the defaults that I can change if necessary. I don't believe deleting them will do anything but I like to leave them there just in case.

**Review by dhgreene — APPROVED** [review]
👍

**Pull request merged**